src/main/java/com/dataguise/saas/service/TaskBuilderService.java

```diff
@@ -0,0 +1,148 @@
+                    detectionTaskBuilder.addPolicyIds(3);
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    detectionTaskBuilder.addPolicyIds(8);
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) detectionTaskBuilder;
+        } else if( taskBuilder instanceof MaskerTaskBuilder) {
+            MaskerTaskBuilder maskerTaskBuilder = ((MaskerTaskBuilder) taskBuilder);
+            for(String policyType : policyList) {
+                if(policyType.equalsIgnoreCase(PolicyType.HIPAA.name())) {
+                    maskerTaskBuilder.addPolicy(1);
+                } else if (policyType.equalsIgnoreCase(PolicyType.PCI.name())) {
+                    maskerTaskBuilder.addPolicy(2);
+                } else if (policyType.equalsIgnoreCase(PolicyType.PII.name())) {
+                    maskerTaskBuilder.addPolicy(3);
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    maskerTaskBuilder.addPolicy(8);
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) maskerTaskBuilder;
+        } else if( taskBuilder instanceof LFADetectionTask.LFADetectionTaskBuilder) {
+            LFADetectionTask.LFADetectionTaskBuilder LFADetectionTaskBuilder = ((LFADetectionTask.LFADetectionTaskBuilder) taskBuilder);
+            for(String policyType : policyList) {
+                if(policyType.equalsIgnoreCase(PolicyType.HIPAA.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("HIPAA_Hadoop");
+                } else if (policyType.equalsIgnoreCase(PolicyType.PCI.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("PCI_Hadoop");
+                } else if (policyType.equalsIgnoreCase(PolicyType.PII.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("PII_Hadoop");
+                } else if (policyType.equalsIgnoreCase(PolicyType.GDPR.name())) {
+                    LFADetectionTaskBuilder.addPolicyName("GDPR_Hadoop");
+                } else {
+                    throw new InternalServerException(ErrorConstants.INVALID_POLICY_TYPE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
+            builder = (T1) LFADetectionTaskBuilder;
+        }
+        return builder;
+    }
+    public String getTimeStampForDgSecure() {
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
+        Date date = new Date();
+        TimeZone tz = TimeZone.getTimeZone("UTC");
+        sdf.setTimeZone(tz);
+        String timeStamp = sdf.format(date);
+
+        logger.info("TIME STAMP IS: " + timeStamp);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/TaskBuilderService.java L123
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1708163650>*